test/e2e/storage/testpatterns/testpattern.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testpatterns
+
+import (
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	MinFileSize = 1 * framework.MiB
+
+	FileSizeSmall  = 1 * framework.MiB
+	FileSizeMedium = 100 * framework.MiB
+	FileSizeLarge  = 1 * framework.GiB
+)
+
+type TestVolType string
+
+var (
+	InlineVolume     TestVolType = "InlineVolume"
+	PreprovisionedPV TestVolType = "PreprovisionedPV"
+	DynamicPV        TestVolType = "DynamicPV"
+)
+
+type TestPattern struct {
+	Name        string
+	VolType     TestVolType
+	FsType      string
+	VolBindMode storagev1.VolumeBindingMode
```

Comment:
The test framework itself doesn't seem to be utilizing VolBindMode or VolMode (it's self-contained in the volumemode test suite).  Can we somehow move these to the volumemode file?